go-controller/pkg/ovn/egressfirewall_test.go

```diff
@@ -984,23 +1044,57 @@ var _ = ginkgo.Describe("OVN EgressFirewall Operations", func() {
 					startOvn(dbSetup, nil, nil)
 
 					var err error
-					setDNSOpsMock(dnsName, resolvedIP)
-					fakeOVN.controller.egressFirewallDNS, err = NewEgressDNS(fakeOVN.controller.addressSetFactory,
-						fakeOVN.controller.controllerName, fakeOVN.controller.stopChan)
-					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					if oldDNS {
+						setDNSOpsMock(dnsName, resolvedIP)
+						fakeOVN.controller.dnsNameResolver, err = dnsnameresolver.NewEgressDNS(fakeOVN.controller.addressSetFactory,
+							fakeOVN.controller.controllerName, fakeOVN.controller.stopChan, egressFirewallDNSDefaultDuration)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					} else {
+						// Initialize the dnsNameResolver.
+						fakeOVN.controller.dnsNameResolver, err = dnsnameresolver.NewExternalEgressDNS(fakeOVN.controller.addressSetFactory,
+							fakeOVN.controller.controllerName, true, fakeOVN.watcher.DNSNameResolverInformer().Informer())
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						err = fakeOVN.controller.dnsNameResolver.Run()
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					}
 
 					_, err = fakeOVN.fakeClient.EgressFirewallClient.K8sV1().EgressFirewalls(egressFirewall.Namespace).
 						Create(context.TODO(), egressFirewall, metav1.CreateOptions{})
 					gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
+					if !oldDNS {
+						dnsNameResolver := newDNSNameResolverObject("dns-default", config.Kubernetes.OVNConfigNamespace, dnsNameLowerCaseFQDN,
+							ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+								ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+									{
+										DNSName: ocpnetworkapiv1alpha1.DNSName(dnsNameLowerCaseFQDN),
+										ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+											{
+												IP: resolvedIP,
+											},
+										},
+									},
+								},
+							})
+
+						// Create the dns name resolver object.
+						_, err = fakeOVN.fakeClient.OCPNetworkClient.NetworkV1alpha1().DNSNameResolvers(dnsNameResolver.Namespace).
+							Create(context.TODO(), dnsNameResolver, metav1.CreateOptions{})
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					}
```

Comment:
I think we can join these 2 pieces and run `EgressFirewall.Create` after it. Then we could have a function like `startDNSResolver(oldDNS, dnsName, resolvedIPs)` that would set up dns mock or create DNSNameResolver depending on the mode to avoid code duplication with the next test